bots/jep/src/main/java/org/openjdk/skara/bots/jep/JEPBot.java

```diff
@@ -0,0 +1,149 @@
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package org.openjdk.skara.bots.jep;
+
+import org.openjdk.skara.bot.Bot;
+import org.openjdk.skara.bot.WorkItem;
+import org.openjdk.skara.forge.HostedRepository;
+import org.openjdk.skara.forge.PullRequest;
+import org.openjdk.skara.issuetracker.IssueProject;
+
+import java.nio.file.Path;
+import java.util.Collection;
+import java.util.List;
+import java.util.logging.Logger;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static org.openjdk.skara.bots.jep.JEPBotFactory.NAME;
+
+public class JEPBot implements Bot, WorkItem {
+    final static String JEP_LABEL = "jep";
+    private final Logger log = Logger.getLogger("org.openjdk.skara.bots");
+    private final static Pattern jepMarkerPattern = Pattern.compile("<!-- jep: '(.*?)' '(.*?)' '(.*?)' -->");
+    private final HostedRepository repo;
+    private final IssueProject issueProject;
+
+    JEPBot(HostedRepository repo, IssueProject issueProject) {
+        this.repo = repo;
+        this.issueProject = issueProject;
+    }
+
+    @Override
+    public boolean concurrentWith(WorkItem other) {
+        if (!(other instanceof JEPBot otherBot)) {
+            return true;
+        }
+        return !repo.isSame(otherBot.repo);
+    }
+
+    @Override
+    public Collection<WorkItem> run(Path scratchPath) {
+        var prs = repo.pullRequests();
+        for (var pr : prs) {
+            var jepComment = pr.comments().stream()
+                    .filter(comment -> comment.author().equals(pr.repository().forge().currentUser()))
+                    .flatMap(comment -> comment.body().lines())
+                    .map(jepMarkerPattern::matcher)
+                    .filter(Matcher::find)
+                    .reduce((first, second) -> second)
+                    .orElse(null);
+            if (jepComment == null) {
+                if (pr.labelNames().contains(JEP_LABEL)) {
+                    pr.removeLabel(JEP_LABEL);
```

Comment:
Missed this remove label in the previous review due to the long thread in the middle of the block. Please add logging here too.

Reply:
I missed it, too. Added. Use the `info` level instead of `fine` because the condition only occurs when the developers add the `jep` label manually in the PR.